elements/pfe-switch/src/pfe-switch.scss

```diff
@@ -0,0 +1,219 @@
+$toggle-knob-diameter:   14px;
+$toggle-knob-offset:  5px;
+
+///
+/// Convert px to em
+///
+/// @param $value {length}  Value to convert with px suffix
+/// @param $base-size {length} Optional, font size to use to calculate em
+/// @return {length} Em version of input
+@function px-to-em($value) {
+  @return $value / $font-size * 1em;
+}
+
+//sass-lint:disable indentation
+// Variables based on PF Next Switch
+$variables: (
+  toggle-knob--default__BackgroundColor: #fff,
+  toggle-knob--disabled__Color: #d2d2d2,
+  toggle-knob__BoxShadow: '0 0.0625rem 0.0625rem 0rem rgba(3, 3, 3, 0.05), 0 0.25rem 0.5rem 0.25rem rgba(3, 3, 3, 0.06)',
+
+  toggle--default__BackgroundColor: #d2d2d2,
+  toggle--checked__BackgroundColor: #06c,
+  toggle--disabled__BackgroundColor: #737679,
+
+  label--default__Color: #151515,
+  label--checked__Color: #737679,
+  label--disabled__Color: #737679,
+);
+//sass-lint:enable indentation
+
+// #{pfe-local(paddingTop)}
+
+:host {
+  @include pfe-set-local($variables);
+  display: inline-block;
+  margin: 0 1.5em 0 0;
+  overflow: hidden;
+}
+
+:host([hidden]) {
+  display: none;
+}
+
+.pfe-switch {
+  position: relative;
+}
+
+::slotted(input) {
+  @include pfe-screen-reader-only;
+}
+
+::slotted(.sr-only),
+.sr-only {
+  text-indent: -9999em;
+}
+
+::slotted(label) {
+  position: relative;
+  display: block;
+  // Width of the switch + some white space
+  padding-left: #{px-to-em($toggle-width) + 0.5em};
+  font-size: px-to-em($font-size);
+  line-height: $line-height / $font-size; // Unitless font-size
+  color: #{pfe-local(label--default__Color)};
+  cursor: pointer;
+
+  // Toggle container
+  &:before {
+    content: '';
+    position: absolute;
+    top: 0;
+    left: 0;
+    display: block;
+    width: px-to-em($toggle-width);
+    height: px-to-em($toggle-height);
+    border-radius: px-to-em($toggle-height / 2);
+    background: #{pfe-local(toggle--default__BackgroundColor)};
+    transition: background-color 0.25s;
+  }
+
+  // Toggle knob
+  &:after {
+    content: '';
+    position: absolute;
+    top: $toggle-knob-offset;
+    left: $toggle-knob-offset;
+    display: block;
+    width: px-to-em($toggle-knob-diameter);
+    height: px-to-em($toggle-knob-diameter);
+    border-radius: 50%;
+    background: #{pfe-local(toggle-knob--default__BackgroundColor)};
+    box-shadow: #{pfe-local(toggle-knob__BoxShadow)};
+    transition: transform 0.25s cubic-bezier(0.21, 0.84, 0.56, 1.55);
+  }
+}
+
+::slotted([checked])  {
+  color: #{pfe-local(label--checked__Color)};
+
+  // Toggle container
+  &:before {
+    background: #{pfe-local(toggle--checked__BackgroundColor)};
+  }
+
+  // Toggle knob
+  &:after {
+    // First transform gets the right bound,
+    // the second moves the knob back behind the edge
+    transform:
+      translate(#{$toggle-width - ($toggle-knob-offset * 2)})
+      translate(-100%);
+  }
+}
+
+::slotted([disabled]) {
+  color: #{pfe-local(label--disabled__Color)};
```

Comment:
Instead of redefining properties when state changes, we've found it more efficient to redefine the custom property value.

```
::slotted(label) {
  color: #{pfe-local($cssvar: Color, $region: label)};
}

::slotted([disabled]) {
  --pfe-switch__label--Color: #{pfe-local(label--disabled__Color)};
}
 ```

Reply:
General template for properties: `--<component name>__<region>--<Property>--<state>`

Reply:
nice!